modules/core/src/main/java/org/apache/ignite/internal/cluster/IgniteClusterImpl.java

```diff
@@ -639,6 +645,64 @@ private boolean changeWalMode(String cacheName, boolean enabled) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public UUID id() {
+        return id;
+    }
+
+    /**
+     * Not part of public API.
+     * Enables ClusterProcessor to set ID in the following cases:
+     * <ol>
+     *     <li>For the first time on node startup.</li>
+     *     <li>Set to null on client disconnect.</li>
+     *     <li>Set to some not-null value on client reconnect.</li>
+     * </ol>
+     *
+     * @param id ID to set.
+     */
+    public void setId(UUID id) {
+        this.id = id;
+    }
+
+    /** {@inheritDoc} */
+    @Override public String tag() {
+        return tag;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void tag(String tag) throws IgniteCheckedException {
+        if (tag == null || tag.isEmpty())
+            throw new IgniteCheckedException("Please provide not-null and not empty string for cluster tag");
+
+        if (tag.length() > MAX_TAG_LENGTH)
+            throw new IgniteCheckedException("Maximum tag length is exceeded, max length is " +
+                MAX_TAG_LENGTH +
```

Comment:
I'm not sure that using that much lines is required here. If it is, CG mandate the use of curly brace here: if body longer than one line.

Reply:
Fixed - added curly braces.